test/miscellaneous_tests/symbolic_stoichiometry.jl

```diff
@@ -1,256 +1,294 @@
 ### Prepares Tests ###
 
 # Fetch packages.
-using Catalyst, JumpProcesses, LinearAlgebra, OrdinaryDiffEq, Test
+using Catalyst, JumpProcesses, OrdinaryDiffEq, StochasticDiffEq, Statistics, Test
+using Symbolics: unwrap
 
 # Sets stable rng number.
 using StableRNGs
 rng = StableRNG(12345)
+seed = rand(rng, 1:100)
 
 # Sets the default `t` to use.
 t = default_t()
 
+# Fetch test functions.
+include("../test_functions.jl")
+
+
 ### Base Tests ###
 
-# Probably requires us to permit Int64 parameters for this to work. 
-@test_broken let
-    @parameters k α
+# Checks that systems with symbolic stoichiometries, created using different approaches, are identical.
+let 
+    @parameters p k d::Float64 n1::Int64 n2 n3
+    @species X(t) Y(t)
+    rxs1 = [
+        Reaction(p, nothing, [X], nothing, [n1])
+        Reaction(k, [X], [Y], [n2], [n3])
+        Reaction(d, [Y], nothing)
+    ]
+    rs1 = ReactionSystem(rxs1, t; name = :rs)
+
+    rxs2 = [
+        @reaction p, 0 --> $n1*X
+        @reaction k, n2*X --> n3*Y
+        @reaction $d, Y --> 0
+    ]
+    rs2 = ReactionSystem(rxs2, t; name = :rs)
+
+    rs3 = @reaction_network rs begin
+        @parameters d::Float64 n1::Int64
+        p, 0 --> n1*X
+        k, n2*X --> n3*Y
+        d, Y --> 0
+    end
+
+    @test rs1 == rs2 == rs3
+    @test issetequal(unknowns(rs1), [X, Y])
+    @test issetequal(parameters(rs1), [p, k, d, n1, n2, n3])
+    @test unwrap(d) isa SymbolicUtils.BasicSymbolic{Float64}
+    @test unwrap(n1) isa SymbolicUtils.BasicSymbolic{Int64}
+end
+
+# Declares a network, parameter values, and initial conditions, to be used for the next couple of tests.
+begin
+    # Prepares parameters and species and their values.
+    @parameters k α::Int64
     @species A(t), B(t), C(t), D(t)
-    rxs = [Reaction(t * k, [A], [B], [2 * α^2], [k + α * C])
-           Reaction(1.0, [A, B], [C, D], [α, 2], [k, α])]
-    @named rs = ReactionSystem(rxs, t)
-    @test issetequal(unknowns(rs), [A, B, C, D])
-    @test issetequal(parameters(rs), [k, α])
-    osys = convert(ODESystem, rs)
-
-    g = (k + α * C)
-    rs2 = @reaction_network rs begin
-        t * k, 2 * α^2 * A --> $g * B
-        1.0, α * A + 2 * B --> k * C + α * D
+    u0_1 = (A => 3.0, B => 2.0, C => 3.0, D => 5.0)
+    ps_1 = (k => 5.0, α => 2)
+    u0_2 = [u[2] for u in u0_1]
+    ps_2 = [p[2] for p in ps_1]
```

Comment:
Yes:
![image](https://github.com/SciML/Catalyst.jl/assets/18099310/2934a86d-b5b6-4c29-aeef-09a5abab91ea)

But the point is good. My actual idea was to mess the test up a bit and use Int for one, and float (that are int) for the other, since MTK supports this now days for stuff like this.
(Happy to just have the second thing be a normal Tuple with Float64/Int64 like you suggested previously though)

Reply:
sounds good, Tuple it is.